var/spack/repos/builtin/packages/cp2k/package.py

```diff
@@ -351,11 +370,27 @@ class Cp2k(MakefilePackage, CudaPackage, CMakePackage, ROCmPackage):
         for arch in CudaPackage.cuda_arch_values:
             if arch not in supported_cuda_arch_list:
                 conflicts("+cuda", when="cuda_arch={0}".format(arch), msg=cuda_msg)
+            depends_on("dla-future cuda_arch={0}".format(arch), when="cuda_arch={0}".format(arch))
+            depends_on("sirius cuda_arch={0}".format(arch), when="cuda_arch={0}".format(arch))
+            depends_on("dbcsr cuda_arch={0}".format(arch), when="cuda_arch={0}".format(arch))
+            depends_on("py-torch cuda_arch={0}".format(arch), when="cuda_arch={0}".format(arch))
 
     with when("+rocm"):
         for arch in ROCmPackage.amdgpu_targets:
             if arch not in supported_rocm_arch_list:
                 conflicts("+rocm", when="amdgpu_target={0}".format(arch), msg=rocm_msg)
+            depends_on(
+                "dla-future amdgpu_target={0}".format(arch), when="amdgpu_target={0}".format(arch)
+            )
+            depends_on(
+                "sirius amdgpu_target={0}".format(arch), when="amdgpu_target={0}".format(arch)
+            )
+            depends_on(
+                "dbcsr amdgpu_target={0}".format(arch), when="amdgpu_target={0}".format(arch)
+            )
+            depends_on(
+                "py-torch amdgpu_target={0}".format(arch), when="amdgpu_target={0}".format(arch)
+            )
```

Comment:
I think the dependencies need to be qualified with `when="... +dep` as well, otherwise they become unconditional dependencies, no? At least `dla-future` needs a `when="+dlaf"`.

Reply:
agreed. It should be fixed now